### PR TITLE
Feature/M2-1955 Add performance tasks

### PR DESCRIPTION
**Pull request by IhorNesterenko99**
Builder - Performance Tasks - add Performance Task button, dropdown, add to the form and list, remove from payload on create/update applet

image

### CODE REVIEW

File: src/modules/Builder/features/Activities/Activities.types.ts [hunk 1]
```diff
@@ -5,4 +5,20 @@ export type GetActivitiesActions = {
...
   onVisibilityChange: () => void;
   onRemove: () => void;
+  isEditVisible: boolean;
 };
+
+export type ActivityAddProps = {
+  index?: number;
+  performanceTaskName?: string;
+  performanceTaskDesc?: string;
+  notNavigate?: boolean;
```
Comment (alex-i-e) [inline src/modules/Builder/features/Activities/Activities.types.ts:15]: maybe to rename it to `shouldBlockNavigation` or `isNavigationBlocked`

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx [hunk 2]
```diff
@@ -175,7 +175,8 @@ export const ActivityFlowBuilder = () => {
             type: GetMenuItemsType.ChangeActivity,
             index: indexToUpdate ?? undefined,
             onMenuClose: () => setAnchorEl(null),
-            activities,
+            // TODO: remove filtering after connecting Performance Tasks API
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:178]: could you add the BE task number if any?

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.utils.tsx [hunk 3]
```diff
@@ -13,13 +17,18 @@ const checkOnIdOrKey = (checkId: string) => (entity: ActivityFlowFormValues) =>
 export const getActivityFlowIndex = (activityFlows: ActivityFlowFormValues[], checkId: string) =>
   activityFlows.findIndex(checkOnIdOrKey(checkId));
 
-export const getActivitiesIdsObjects = (activities: ActivityFormValues[]) =>
-  activities.reduce((acc: Record<string, ActivityFormValues>, activity) => {
-    const id = activity.id || activity.key || '';
-    acc[id] = activity;
+export const getActivitiesIdsObjects = (
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.utils.tsx:20]: please use `getObjectFromList` in shared/utils

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilderHeader/ActivityFlowBuilderHeader.tsx [hunk 4]
```diff
@@ -48,7 +48,8 @@ export const ActivityFlowBuilderHeader = ({
             menuItems={getMenuItems({
               type: GetMenuItemsType.AddActivity,
               onMenuClose: () => setAnchorEl(null),
-              activities,
+              // TODO: remove filtering after connecting Performance Tasks API
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilderHeader/ActivityFlowBuilderHeader.tsx:51]: could you add the BE task number if any?

File: src/modules/Builder/features/Activities/Activities.tsx [hunk 5]
```diff
@@ -136,57 +153,71 @@ export const Activities = () => {
...
-                          onModalClose={handleModalClose}
-                          onModalSubmit={() => handleActivityRemove(index, activityKey)}
-                        />
-                      </Fragment>
-                    );
-                  })}
+                  {activities.map(
+                    (
+                      activity:
+                        | ActivityFormValues
```
Comment (kdanilenka32) [inline src/modules/Builder/features/Activities/Activities.tsx:159]: pls move to types

File: src/modules/Builder/features/Activities/Activities.utils.tsx [hunk 6]
```diff
@@ -1,32 +1,41 @@
...
+  const actions = [];
+  if (isEditVisible) {
+    actions.push({
+      icon: <Svg id="edit" />,
+      action: onEdit,
+    });
+  }
+
+  return [
+    ...actions,
```
Comment (kdanilenka32) [inline src/modules/Builder/features/Activities/Activities.utils.tsx:26]: ...(isEditVisible
    ? [ {
      icon: <Svg id="edit" />,
      action: onEdit,
    }]: []

File: src/modules/Builder/features/Activities/ActivitiesHeader/ActivitiesHeader.utils.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+const { t } = i18n;
+
+export const getPerformanceTasksMenu = (
+  onAddActivity: (props: ActivityAddProps) => void,
+  setAnchorEl: (el: null | HTMLElement) => void,
+) => {
+  const getAction = (props: ActivityAddProps) => {
+    const { performanceTaskName, performanceTaskDesc, isNavigationBlocked } = props || {};
+    onAddActivity({ performanceTaskName, performanceTaskDesc, isNavigationBlocked });
```
Comment (kdanilenka32) [inline src/modules/Builder/features/Activities/ActivitiesHeader/ActivitiesHeader.utils.tsx:13]: y can pass just onAddActivity({ ...props, , isNavigationBlocked: true});

File: src/modules/Builder/features/Activities/Activities.tsx [hunk 8]
```diff
@@ -136,57 +153,71 @@ export const Activities = () => {
...
+                                <Item
+                                  {...activity}
+                                  onItemClick={() => handleEditActivity(index)}
+                                  dragHandleProps={itemProvided.dragHandleProps}
+                                  isDragging={snapshot.isDragging}
+                                  img={activity.image}
+                                  isInactive={activity.isHidden}
+                                  hasStaticActions={activity.isHidden}
+                                  uiType={ItemUiType.Activity}
+                                  getActions={() =>
```
Comment (kdanilenka32) [inline src/modules/Builder/features/Activities/Activities.tsx:186]: It should be a func?

File: src/modules/Builder/features/Activities/Activities.utils.tsx (hunk 6 above)
Comment (IhorNesterenko99) [reply src/modules/Builder/features/Activities/Activities.utils.tsx:26]: done

**Review by kdanilenka32 — APPROVED** [review]

**Pull request merged**